File: jupyter2/ps/iteration/gcd/jl/gcd.jl
### begin my answer

### begin hidden
function gcd(a, b)
    while b != 0
        a, b = b, a % b
    end
    a
end
### end hidden
### end my answer

function main()
    @assert gcd(1499276220, 463728183) == 6873
    @assert gcd(256381708674, 48941846742) == 35094
    @assert gcd(8619803849, 3861314192) == 11437
    println("OK")
end

main()

File: jupyter2/ps/iteration/gcd/rs/gcd.rs
/** begin my answer */

/** begin hidden */
fn gcd(a: i64, b: i64) -> i64 {
    let mut x = a;
    let mut y = b;
    while y != 0 {
        let r = x % y;
        x = y;
        y = r;
    }
    x
}
/** end hidden */
/** end my answer */

fn main() {
    assert!(gcd(1499276220, 463728183) == 6873);
    assert!(gcd(256381708674, 48941846742) == 35094);
    assert!(gcd(8619803849, 3861314192) == 11437);
    println!("OK");
}
